src/etc/inc/system.inc

```diff
@@ -328,7 +328,7 @@ function get_searchdomains()
         $master_list[] = $syscfg['dnssearchdomain'];
     }
 
-    if (isset($syscfg['dnsallowoverride'])) {
+    if (!empty($syscfg['dnsallowoverride'])) {
```

Comment:
we did have issues with one of these before

Reply:
@fichtner not sure what it was, but for simplicity, I prefer the `empty()` variants, so if it doesn't work as it should, we might have to figure out why that is.

Reply:
this was https://github.com/opnsense/core/issues/2751 and looks like it was fixed in the way we want, but then we should move all spots to !empty, not just this one

Reply:
@fichtner let's change the rest of them as well indeed, https://github.com/opnsense/core/commit/588566fba8eb5eb170777e22a4b7fd51c5ea952a should do the trick.